pennylane_lightning/core/src/utils/NDPermuter.hpp

```diff
@@ -242,7 +244,7 @@ template <size_t BLOCKSIZE = 1024> class DefaultPermuter {
 
                 bool complete{true};
                 // NOLINTBEGIN
-                for (size_t j = num_indices; j--;) {
+                for (std::size_t j = num_indices; j--;) {
```

Comment:
Is `j==0` the stop condition here?

Reply:
The stop condition is `++old_counter[j] < shape[j]` or `j==0` whichever satisfies sooner.